connect/api/src/main/java/org/apache/kafka/connect/sink/SinkRecord.java

```diff
@@ -100,4 +110,29 @@ public String toString() {
                 ", timestampType=" + timestampType +
                 "} " + super.toString();
     }
+
+    public class InternalSinkRecord extends SinkRecord {
+
+        ConsumerRecord<byte[], byte[]> originalRecord;
+
+        public InternalSinkRecord(String topic, int partition, Schema keySchema, Object key,
+                                  Schema valueSchema, Object value, long kafkaOffset,
+                                  Long timestamp, TimestampType timestampType,
+                                  Iterable<Header> headers,
+                                  ConsumerRecord<byte[], byte[]> originalRecord) {
+            super(topic, partition, keySchema, key, valueSchema, value, kafkaOffset, timestamp,
+                timestampType, headers);
+            this.originalRecord = originalRecord;
+
+        }
+
+        /**
+         *
+         * @return the original consumer record that was converted to this sink record.
+         */
```

Comment:
If we're going to add JavaDoc, which I think is helpful, then make it complete by adding a description:
```suggestion
        /**
         * Return the original consumer record that this sink record represents.
         *
         * @return the original consumer record; never null
         */
```